<launch>


  <!-- This is the outer launch file for mongodb_store. It sets a machine tag for the mongodb_store_inc.launch to use. If you already have a machine tag defined, you can call mongodb_store_inc.launch directly and use the machine argument to it.-->

  <arg name="db_path" default="/var/local/mongodb_store"/>
  <arg name="port" default="62345" />
  <arg name="defaults_path" default=""/>
  <arg name="replicator_dump_path" default="/tmp/replicator_dumps"/>
  <arg name="use_daemon" default="false" />


  <arg name="machine" default="localhost" />
  <arg name="user" default="" />
  <arg name="test_mode" default="false" />
  <arg name="use_repl_set" default="false" />
  <arg name="repl_set" default="rs0" />
  <arg name="queue_size" default="100" />
  <arg name="bind_to_host" default="false" />

  <arg name="use_localdatacenter" default="true" />

  <machine name="$(arg machine)" address="$(arg machine)" env-loader="$(optenv ROS_ENV_LOADER )" user="$(arg user)"/>

  <include file="$(find mongodb_store)/launch/mongodb_store_inc.launch">
    <arg name="db_path" value="$(arg db_path)"/>
    <arg name="port" value="$(arg port)"/>
    <arg name="defaults_path" value="$(arg defaults_path)"/>
    <arg name="replicator_dump_path" value="$(arg replicator_dump_path)"/>
    <arg name="use_daemon" value="$(arg use_daemon)"/>
    <arg name="machine" value="$(arg machine)"/>    
    <arg name="test_mode" value="$(arg test_mode)"/>
    <arg name="use_repl_set" value="$(arg use_repl_set)"/>
    <arg name="repl_set" value="$(arg repl_set)"/>
    <arg name="queue_size" value="$(arg queue_size)" />
    <arg name="bind_to_host" value="$(arg bind_to_host)" />
  </include>

</launch>
